backend/app/db/crud/document.py

```diff
@@ -155,11 +157,27 @@ def __init__(self, category: str) -> None:
         )
 
 
+def _create_document_slug(
+    document_name: str,
+    geography: Geography,
+    publication_ts: datetime,
+) -> str:
+    return "_".join(
+        [
+            slugify(f"{geography.display_value}", separator="-"),
+            f"{publication_ts.year}",
+            slugify(document_name, separator="-"),
+        ]
+    )[
+        :1024
+    ]  # just limit the slug to a reasonable length
```

Comment:
We need some sort of mechanism to check/retry for non-uniquess

Reply:
you're right, and we should track that, but for now I've used the unique components of the import id to cheat a bit